src/entity/projectile/Arrow.php

```diff
@@ -174,13 +174,15 @@ public function onCollideWithPlayer(Player $player) : void{
 		}
 
 		$item = VanillaItems::ARROW();
-
 		$playerInventory = $player->getInventory();
-		if($player->hasFiniteResources() and !$playerInventory->canAddItem($item)){
-			return;
+		if(!$playerInventory->canAddItem($item)){
+			if($player->hasFiniteResources()){
+				return;
```

Comment:
I think it would be better to pre-cancel the event instead of hardcoding this behaviour. This would allow plugins to customise what happens when a player tries to pickup an item with a full inventory (e.g. sending it directly to their ender chest inventory instead).